source/common/network/address_impl.cc

```diff
@@ -183,10 +183,9 @@ int Ipv4Instance::connect(int fd) const {
 
 int Ipv4Instance::socket(SocketType type) const { return socketFromSocketType(type); }
 
-std::array<uint8_t, 16> Ipv6Instance::Ipv6Helper::address() const {
-  std::array<uint8_t, 16> result;
-  std::copy(std::begin(address_.sin6_addr.s6_addr), std::end(address_.sin6_addr.s6_addr),
-            std::begin(result));
+absl::uint128 Ipv6Instance::Ipv6Helper::address() const {
+  absl::uint128 result{0};
+  memcpy(&result, &address_.sin6_addr.s6_addr, 16);
```

Comment:
I would add a static_assert everywhere you do memcpy to make sure sizeof(absl::uint128) is what you expect, and then potentially use sizeof(absl::uint128) instead of 16.